Add static analysis for all communication related projects.

codito opened the pull request:
Update nuget.config use packages directory relative to sln.

(No functionality changes in this PR)

Commit:
Add static analysis for all communication related projects.

Update nuget.config use packages directory relative to sln.

msftclas commented:
@codito,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

mayankbansal018 commented on src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs:
```diff
@@ -34,7 +34,7 @@ private JsonDataSerializer()
                                     DateParseHandling = DateParseHandling.DateTimeOffset,
                                     DateTimeZoneHandling = DateTimeZoneHandling.Utc,
-                                    TypeNameHandling = TypeNameHandling.None
+                                    TypeNameHandling = TypeNameHandling.None,
```
Additional comma?

mayankbansal018 commented on src/package/VSIXProject/source.extension.vsixmanifest:
```diff
@@ -1,7 +1,7 @@
 <PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
   <Metadata>
-    <Identity Id="Microsoft.TestPlatform.VSIX.Microsoft.0e51abf6-9f5f-401e-a812-4043b87acabf" Version="$version$" Language="en-US" Publisher="Microsoft" />
+    <Identity Id="Microsoft.TestPlatform.VSIX.Microsoft.0e51abf6-9f5f-401e-a812-4043b87acabf" Version="15.1.0" Language="en-US" Publisher="Microsoft" />
```
Is this intentional?

Commit:
Move building vsix to artifacts directory since it modifies a file during build.
Move dependency on roslyn build tools to dependencies props.
Disable CS2008 (no source) warning for vsix and package projects.
